### ISSUE TITLE
[v3] bottom sheet does not react to after close is called consecutively

### ISSUE THREAD

**Issue by cltsang** [gorhom/react-native-bottom-sheet#374]
# Bug

On v3 branch, bottom sheet does not react to any calls after close is called consecutively.

## Environment info

| Library                         | Version |
| ------------------------------- | ------- |
| @gorhom/bottom-sheet         | 3.6.0   |
| react-native                    |0.64.0   |
| react-native-reanimated         |2.0.1   |
| react-native-gesture-handler    |1.10.3   |

## Steps To Reproduce

1. Open either one of the examples in "BASIC", "ADVANCED", or "React Navigation".
2. Press "Close". After close animation has completed, press "Close" once more .
3. Press any button to trigger bottom sheet

Describe what you expected to happen:

1. Bottom sheet should be able to expand, or react to method calls after consecutive close calls
2.

## Reproducible sample code

I just cloned the current example v3 branch (https://github.com/gorhom/react-native-bottom-sheet/commit/0fd2b9d02ce91e635eacc5ccda115eacb7c6a8be) and ran npm install. Tested on iOS simulator.

It seems after `isClosing.current` is set to true [here](https://github.com/gorhom/react-native-bottom-sheet/blob/0fd2b9d02ce91e635eacc5ccda115eacb7c6a8be/src/components/bottomSheet/BottomSheet.tsx#L531), nowhere can it be reset to false.

Thank you.

**Comment by cltsang** [comment]
It seems callback in [`animate`](https://github.com/gorhom/react-native-bottom-sheet/blob/0fd2b9d02ce91e635eacc5ccda115eacb7c6a8be/src/utilities/animate.ts) is not called when animation is not run

**Comment by cltsang** [comment]
So it seems fine after adding these few lines
```js
        if (animatedPosition.value === point) {
          animateToPointCompleted();
          isClosing.current = false;
        }
```
at [line 402](https://github.com/gorhom/react-native-bottom-sheet/blob/0fd2b9d02ce91e635eacc5ccda115eacb7c6a8be/src/components/bottomSheet/BottomSheet.tsx#L402).

**Comment by gorhom** [comment]
thanks @cltsang for reporting this bug ! it should be resolved with `v3.6.1` 👏
